### PR TITLE
[nrf noup] boards: Add non secure target for nrf54L15dk

### PR DESCRIPTION
**Pull request by Vge0rge**
This adds the nrf54l15dk/nrf54l15/cpuapp/ns board variant to sdk-zephyr. It allows to build nRF54l15dk with TF-M in nRF Connect SDK.
    
This is done as a noup as the out-of-tree board extension feature is not available. When zephyrproject-rtos/zephyr#69548 is completed this commit can be reverted and the variant be placed in sdk-nrf.

### CODE REVIEW

File: boards/nordic/nrf54l15dk/board.cmake [hunk 1]
```diff
@@ -7,5 +7,14 @@ elseif(CONFIG_SOC_NRF54L15_CPUFLPR)
...
 
+if(BOARD_NRF54L15DK_NRF54L15_CPUAPP_NS)
+  set(TFM_PUBLIC_KEY_FORMAT "full")
+endif()
+
+if(CONFIG_TFM_FLASH_MERGED_BINARY)
+  set_property(TARGET runners_yaml_props_target PROPERTY hex_file tfm_merged.hex)
+endif()
+
+include(${ZEPHYR_BASE}/boards/common/nrfjprog.board.cmake)
```
Comment (tomi-font) [inline boards/nordic/nrf54l15dk/board.cmake:18]:
```suggestion
```
(No) thanks for the nice little trap. :wink:

**Review by tomi-font — CHANGES_REQUESTED** [review]
Why isn't this marked as a `noup`?

**Review by tomi-font — APPROVED** [review]

**Comment by tomi-font** [comment]
Can someone from the reviewer list review this PR?

File: boards/nordic/nrf54l15dk/doc/index.rst [hunk 2]
```diff
@@ -98,7 +98,7 @@ to be build as multicore configuration with code snippet called ``vpr_launcher``
 for the application core.
 
 Enter the following command to compile ``hello_world`` for the FLPR core::
- west build -p -b nrf54l15pdk/nrf54l15/cpuflpr --sysbuild -- -DSB_VPR_LAUNCHER=y
+ west build -p -b nrf54l15dk/nrf54l15/cpuflpr --sysbuild -- -DSB_VPR_LAUNCHER=y
```
Comment (anangl) [inline boards/nordic/nrf54l15dk/doc/index.rst:101]: This should be corrected upstream, not in this noup commit.

**Review by anangl — CHANGES_REQUESTED** [review]

File: boards/nordic/nrf54l15dk/doc/index.rst (hunk 2 above)
Comment (tomi-font) [reply boards/nordic/nrf54l15dk/doc/index.rst]: True.

**Review by anangl — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**